src/common/url.h

```diff
@@ -7,8 +7,10 @@
 
 #include <string>
 
-using UrlDecodeFn = std::string(const std::string& url_encoded);
-UrlDecodeFn urlDecode;
-extern UrlDecodeFn* const URL_DECODE;
+/* Decode a URL.
+ *
+ * Notably this implementation does not decode a '+' to a ' '.
```

Comment:
would it make sense to mention the `&` as well?

Reply:
Are these commonly decoded as a space as well? I couldn't find evidence for this but I didn't spend much time on it.

Reply:
Not exactly: https://github.com/libevent/libevent/blob/release-2.1.12-stable/http.c#L3186-L3188

Reply:
Hm, the code you are linking to is part of how libevent handles the `+`. Or did you want to refer to `?`? I think if it's not very common for `&` to be decoded as space then we don't need to mention it.

Reply:
? seems to be a control char, `If -1, when true we transform plus to space only after we've seen a ?.  -1 is deprecated.`, no need to mention it, thanks for checking.